tests/handlers/auth_test.py

```diff
@@ -0,0 +1,358 @@
+        new_token,
+        keypair.public_key_as_pem(),
+        algorithms=ALGORITHM,
+        audience="https://example.com/",
+    )
+    assert decoded_token == {
+        "act": {
+            "aud": "https://test.example.com/",
+            "iss": "https://orig.example.com/",
+            "jti": ticket.as_handle("oauth2_proxy"),
+        },
+        "aud": "https://example.com/",
+        "email": "some-user@example.com",
+        "exp": ANY,
+        "iat": ANY,
+        "isMemberOf": [{"name": "admin"}],
+        "iss": "https://test.example.com/",
+        "jti": ticket.as_handle("oauth2_proxy"),
+        "scope": "exec:admin read:all",
+        "sub": "some-user",
+        "uid": "some-user",
+        "uidNumber": "1000",
+    }
+    now = time.time()
+    exp_minutes = app["jwt_authorizer/config"].issuer.exp_minutes
+    expected_exp = now + exp_minutes * 60
+    assert expected_exp - 5 <= decoded_token["exp"] <= expected_exp + 5
+    assert now - 5 <= decoded_token["iat"] <= now + 5
+
+    redis_client = app["jwt_authorizer/redis"]
+    session_store = SessionStore("oauth2_proxy", session_secret, redis_client)
+    session = await session_store.get_session(ticket)
+    assert session
+    assert session.token == new_token
+    assert session.user == "some-user@example.com"
+
+
+async def test_authnz_token_reissue_internal(
+    aiohttp_client: TestClient,
+) -> None:
+    """Test requesting token reissuance to an internal audience."""
+    keypair = RSAKeyPair()
+    token = create_test_token(keypair, ["admin"])
+    session_secret = os.urandom(16)
+    app = await create_test_app(keypair, session_secret)
+    client = await aiohttp_client(app)
+
+    r = await client.get(
+        "/auth",
+        params={
+            "capability": "exec:admin",
+            "audience": "https://example.com/api",
+        },
+        headers={"Authorization": f"Bearer {token}"},
+    )
+    assert r.status == 200
+    new_token = r.headers["X-Auth-Request-Token"]
+    assert token != new_token
+    ticket = Ticket.from_str(
+        "oauth2_proxy", r.headers["X-Auth-Request-Token-Ticket"]
+    )
+
+    assert jwt.get_unverified_header(new_token) == {
+        "alg": ALGORITHM,
+        "typ": "JWT",
+        "kid": "some-kid",
+    }
+
+    decoded_token = jwt.decode(
+        new_token,
+        keypair.public_key_as_pem(),
+        algorithms=ALGORITHM,
+        audience="https://example.com/api",
+    )
+    assert decoded_token == {
+        "act": {
+            "aud": "https://example.com/",
+            "iss": "https://test.example.com/",
+            "jti": "some-unique-id",
+        },
+        "aud": "https://example.com/api",
+        "email": "some-user@example.com",
+        "exp": ANY,
+        "iat": ANY,
+        "isMemberOf": [{"name": "admin"}],
+        "iss": "https://test.example.com/",
+        "jti": ticket.as_handle("oauth2_proxy"),
+        "sub": "some-user",
+        "uid": "some-user",
+        "uidNumber": "1000",
```

Comment:
Since you've got this token here and in the test above and it's got just static data might want to put that in a function as what the default test token representation looks like

Reply:
This is a good idea. I'm going to defer this to a later PR because there's a bunch of stuff stacked up, but I will add refactoring the test logic to my todo. I'm not really happy about how magical all these test parameters are to the tests right now.